pkg/quotes/uniswap.go

```diff
@@ -49,22 +70,28 @@ func (u *uniswapV3) Subscribe(market Market) error {
 		return fmt.Errorf("market %s does not exist", symbol)
 	}
 
-	u.mu.Lock()
-	defer u.mu.Unlock()
-	u.streams[market] = make(chan struct{}, 1)
+	u.streams.Store(market, make(chan struct{}, 1))
 
 	go func() {
 		from := time.Now()
 		for {
+			if stream, ok := u.streams.Load(market); ok {
+				stopCh := stream.(chan struct{})
+				select {
+				case <-stopCh:
+					logger.Infof("market %s is stopped", symbol)
+					return
+				default:
```

Comment:
do we need an empty default case here and there?

Reply:
We do, or else `select` will block until `stopCh` is read. The idea here is to read the stop token only if channel is not empty, otherwise continue normal flow.